src/srkf.jl

```diff
@@ -14,6 +14,17 @@ function calc_upper_triangular_of_qr!(A)
     R
 end
 
+"""
+    correct_cholesky_sign(R)
+
+    Return a modify the Cholesky factor `R` with positive diagonals.
+"""
+correct_cholesky_sign(R) = sign.(diag(R)) .* R
+
+function correct_cholesky_sign!(R)
+    R .= sign.(diag(R)) .* R
+end
```

Comment:
This still allocates.
I asked for help with this and the Julia community got crazy again ;)
See https://discourse.julialang.org/t/why-does-a-sign-view-a-diagind-a-a-allocate/122332
So my suggestion is:
```suggestion
function correct_cholesky_sign!(R)
    for i in axes(R,1)
        if sign(R[i,i]) < 0
            R[i, :] = -R[i, :]
        end
    end
    return R
end
```